bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/graphics/ImageLoader.java

```diff
@@ -159,11 +162,61 @@ void reset() {
  * </ul>
  */
 public ImageData[] load(InputStream stream) {
+	return loadDefault(stream);
+}
+
+/**
+ * Loads an array of <code>ImageData</code> objects from the
+ * specified input stream. If the stream is a SVG File and zoom is not 0,
+ * this method will try to rasterize the SVG.
+ * Throws an error if either an error occurs while loading the images, or if the images are not
+ * of a supported type. Returns the loaded image data array.
+ *
+ * @param stream the input stream to load the images from
+ * @param zoom the zoom factor to apply when rasterizing a SVG.
+ * A value of 0 means that the standard method for loading should be used.
+ * @return an array of <code>ImageData</code> objects loaded from the specified input stream
+ *
+ * @exception IllegalArgumentException <ul>
+ *    <li>ERROR_NULL_ARGUMENT - if the stream is null</li>
+ * </ul>
+ * @exception SWTException <ul>
+ *    <li>ERROR_IO - if an IO error occurs while reading from the stream</li>
+ *    <li>ERROR_INVALID_IMAGE - if the image stream contains invalid data</li>
+ *    <li>ERROR_UNSUPPORTED_FORMAT - if the image stream contains an unrecognized format</li>
+ * </ul>
+ *
+ * @since 3.129
+ */
+public ImageData[] load(InputStream stream, int zoom) {
+	if (stream == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
+    reset();
+    if (!stream.markSupported()) {
+		stream = new BufferedInputStream(stream);
+	}
+	SVGRasterizer rasterizer = SVGRasterizerRegistry.getRasterizer();
+	if (rasterizer != null && zoom != 0) {
+		try {
+			if (rasterizer.isSVGFile(stream)) {
+				float scalingFactor = zoom / 100.0f;
+				ImageData rasterizedData = rasterizer.rasterizeSVG(stream, scalingFactor);
+				if (rasterizedData != null) {
+					data = new ImageData[]{rasterizedData};
+				    return data;
+				}
+			}
+		} catch (IOException e) {
+			//ignore.
```

Comment:
Is there a specific reason why for Linux and mac this error is ignored, while it is propagated in Windows?
Usually it's better to not sallow exceptions/errors as it makes it harder to become aware of them and understand them.